src/ArmTemplates/Creator/TemplateCreators/TagTemplateCreator.cs

```diff
@@ -22,53 +24,71 @@ public TagTemplateCreator(ITemplateBuilder templateBuilder)
             this.templateBuilder = templateBuilder;
         }
 
-        public Template CreateTagTemplate(CreatorParameters creatorConfig)
+        static void AddTagNameToDictionary(string tagName, Dictionary<string, string> tagsDictionary)
         {
-            // create empty template
-            Template tagTemplate = this.templateBuilder.GenerateEmptyTemplate().Build();
+            var resourceName = NamingHelper.GenerateValidResourceNameFromDisplayName(tagName);
 
-            // add parameters
-            tagTemplate.Parameters = new Dictionary<string, TemplateParameterProperties>
+            if (string.IsNullOrEmpty(resourceName)) 
             {
-                {ParameterNames.ApimServiceName, new TemplateParameterProperties(){ Type = "string" }}
-            };
+                throw new EmptyResourceNameException(tagName);
```

Comment:
As was discussed, it will be a good approach to have a local validator for the input parameters, so the error could be found before the actual deployment. This kind of validations are helpful for the **creator**